apps/web/app/(roles)/vaga/[id]/RolePage.tsx

```diff
@@ -0,0 +1,285 @@
+'use client'
+
+import React, { useEffect, useState, useCallback, useMemo } from 'react'
+import { FocusBanner } from 'app/landing-page/FocusBanner'
+import SanitizedHTML from 'app/components/SanitizedHTML'
+import {
+  ExternalLink,
+  MapPin,
+  AlarmClock,
+  Laptop,
+  Link,
+  Check,
+  DollarSign,
+  Calendar,
+  Share2,
+} from 'lucide-react'
+
+const COPY_TIMEOUT = 2000
+
+export const RolePage = ({ vaga }) => {
+  const [showShareMenu, setShowShareMenu] = useState(false)
+  const [isCopied, setIsCopied] = useState(false)
+  const shareMenuRef = React.useRef(null)
+
+  const isEnglish = useMemo(() => vaga.language === 'English', [vaga.language])
+
+  const shareButtonText = useMemo(
+    () =>
+      isCopied
+        ? isEnglish
+          ? 'Copied'
+          : 'Copiado'
+        : isEnglish
+        ? 'Copy link'
+        : 'Copiar link',
```

Comment:
Never use nested ternaries, please create a function with `early return` approach for this case.